aequilibrae/paths/route_choice.py

```diff
@@ -0,0 +1,293 @@
+import numpy as np
+import socket
+from aequilibrae.context import get_active_project
+from aequilibrae.paths.graph import Graph, _get_graph_to_network_mapping
+from aequilibrae.paths.route_choice_set import RouteChoiceSet
+from aequilibrae.matrix import AequilibraeMatrix, AequilibraeData
+from typing import Optional, Union, Tuple, List
+import pyarrow as pa
+import pathlib
+import itertools
+
+import logging
+
+
+class RouteChoice:
+    all_algorithms = ["bfsle", "lp", "link-penalisation", "link-penalization"]
+    default_paramaters = {
+        "beta": 1.0,
+        "theta": 1.0,
+        "penalty": 1.1,
+        "seed": 0,
+        "max_routes": 0,
+        "max_depth": 0,
+    }
+
+    def __init__(self, graph: Graph, matrix: AequilibraeMatrix, project=None):
+        self.paramaters = self.default_paramaters.copy()
+
+        proj = project or get_active_project(must_exist=False)
+        self.project = proj
+
+        self.logger = proj.logger if proj else logging.getLogger("aequilibrae")
+
+        self.cores: int = 0
+        self.graph = graph
+        self.matrix = matrix
+        self.__rc = None
+
+        self.schema = RouteChoiceSet.schema
+        self.psl_schema = RouteChoiceSet.psl_schema
+
+        self.compact_link_loads: Optional[np.array] = None
+        self.link_loads: Optional[np.array] = None
+        self.results: Optional[pa.Table] = None
+        self.where: Optional[pathlib.Path] = None
+
+        self.nodes = Optional[Union[List[int], List[Tuple[int, int]]]] = None
+
+    def set_algorithm(self, algorithm: str):
```

Comment:
When we talk about route choice, the algorithm in question is usually the one taking care of the route overlap itself (e.g. Path-Size Logit, Remulla, C-Logit, etc.), so let's change this to **set_choice_set_generation** and merge *set_parameters* in there with it, as that's all about the same thing.